docs/guides/writing_stubs.rst

```diff
@@ -106,3 +106,487 @@ particularly with false positives.
+    class AnotherError(Exception): ...
+
+No::
+
+    def time_func() -> None: ...
+
+    def date_func() -> None: ...  # do no leave unnecessary empty lines
+
+    def ip_func() -> None: ...
+
+
+    class Foo:  # leave only one empty line above
+        x: int
+    class MyError(Exception): ...  # leave an empty line between the classes
+
+Module Level Attributes
+-----------------------
+
+Do not unnecessarily use an assignment for module-level attributes.
+
+Yes::
+
+    CONST: Literal["const"]
+    x: int
+    y: Final = 0  # this assignment conveys additional type information
+
+No::
+
+    CONST = "const"
+    x: int = 0
+    y: float = ...
+    z = 0  # type: int
+    a = ...  # type: int
+
+.. _stub-style-classes:
+
+Classes
+-------
+
+Classes without bodies should use the ellipsis literal ``...`` in place
+of the body on the same line as the class definition.
+
+Yes::
+
+    class MyError(Exception): ...
+
+No::
+
+    class MyError(Exception):
+        ...
+    class AnotherError(Exception): pass
+
+Instance attributes and class variables follow the same recommendations as
+module level attributes:
+
+Yes::
+
+    class Foo:
+        c: ClassVar[str]
+        x: int
+
+    class Color(Enum):
+        # An assignment with no type annotation is a convention used to indicate
+	# an enum member.
+        RED = 1
+
+No::
+
+    class Foo:
+        c: ClassVar[str] = ""
+        d: ClassVar[int] = ...
+        x = 4
+        y: int = ...
+
+Functions and Methods
+---------------------
+
+For keyword-only and positional-or-keyword arguments, use the same
+argument names as in the implementation, because otherwise using
+keyword arguments will fail.
+
+For default values, use the literal values of "simple" default values (``None``,
+bools, ints, bytes, strings, and floats). Use the ellipsis literal ``...`` in
+place of more complex default values. Use an explicit ``X | None`` annotation
+when the default is ``None``.
+
+Yes::
+
+    def foo(x: int = 0) -> None: ...
+    def bar(y: str | None = None) -> None: ...
+
+No::
+
+    def foo(x: X = X()) -> None: ...
+    def bar(y: str = None) -> None: ...
+
+Do not annotate ``self`` and ``cls`` in method definitions, except when
+referencing a type variable.
+
+Yes::
+
+    _T = TypeVar("_T")
+
+    class Foo:
+        def bar(self) -> None: ...
+        @classmethod
+        def create(cls: type[_T]) -> _T: ...
+
+No::
+
+    class Foo:
+        def bar(self: Foo) -> None: ...
+        @classmethod
+        def baz(cls: type[Foo]) -> int: ...
+
+The bodies of functions and methods should consist of only the ellipsis
+literal ``...`` on the same line as the closing parenthesis and colon.
+
+Yes::
+
+    def to_int1(x: str) -> int: ...
+    def to_int2(
+        x: str,
+    ) -> int: ...
+
+No::
+
+    def to_int1(x: str) -> int:
+        return int(x)
+    def to_int2(x: str) -> int:
+        ...
+    def to_int3(x: str) -> int: pass
+
+.. _private-definitions:
+
+Private Definitions
```

Comment:
Duplicates a previous section

Reply:
I consolidated this with the "Stub-Only Objects" section.